### ISSUE TITLE
How can I obtain the drydep and WetLoss of Silicon species in the diagnostic result?

### ISSUE THREAD

**Issue by herizonarivelo** [geoschem/geos-chem#2491]
### Your name

Herizo Narivelo

### Your affiliation

Laboratoire d'Aérologie - Université Toulouse III

### What happened? What did you expect to happen?

Dear Geos-Chem community, 

For my project, I would like to have in my diagnostic output the dry and wet deposition of these two species: SiF1 and SiF2 which are found in species_database.yml as shown in this link: 
https://github.com/geoschem/geos-chem/blob/main/run/shared/species_database.yml 

They are not initially listed in my HISTORY.rc file and I modified the HISTORY.rc file by putting DryDep_SiF1, DryDep_SiF2, WetLossLS_SiF1 and WetLossLS_SiF2 so that I could have the dry and wet deposits of these 2 species in the output. However, when I ran a simulation test; I encountered errors and my simulation aborted. I suspect this is due to the SiF1 and SiF2 species not present in GEOS-Chem. 

I'd like to understand the errors and your help in solving my problem so that I may get both dry and wet deposits in my diagnostic output. And also whether I might be able to get the concentrations of these 2 species. 

I upload below the HISTORY.rc and the GC.log as well as my slurm for my test simulation. In the slurm, I found that SPC_SiF1 and SPC_SiF2 are not in gchp_restart.nc4. So I'm wondering if this is the reason for my problem or something else? If so where do I find the restart with them ? 

Thanks in advance for your answer and your help!

Kind regards, 
Herizo Narivelo

### What are the steps to reproduce the bug?

The bug occurs when I try to modify the HISTORY file by adding the diagnostics I need in my output. 

### Please attach any relevant configuration and log files.

[HISTORY.rc.txt](https://github.com/user-attachments/files/17165780/HISTORY.rc.txt)
[slurm-7933482.out.txt](https://github.com/user-attachments/files/17165783/slurm-7933482.out.txt)
[gchp.20060101_0000z.log.txt](https://github.com/user-attachments/files/17165784/gchp.20060101_0000z.log.txt)

### What GCHP version were you using?

14.2.2

### What environment were you running GCHP on?

Local cluster

### What compiler and version were you using?

gcc 10.2.0

### What MPI library and version were you using?

openmpi-4.0

### Will you be addressing this bug yourself?

Yes, but I will need some help

**Comment by yantosca** [comment]
Thanks for writing @herizonarivelo.  The `SiF1` and `SiF2` species belong to the metals simulation in GEOS-Chem Classic.  We have not ported this option to GCHP at this time.  However, you could add these species into your GCHP simulation.

I would recommend creating a run directory for the GEOS-Chem Classic metals simulation, and then to copy the relevant lines from `HEMCO_Config.rc` and `HEMCO_Diagn,.rc` to the corresponding files in your GCHP run directory.  You will also need to add corresponding entries in `ExtData.rc`.

We have not had a lot of requests to use the metals simulation, which is why we haven't as yet ported it to GCHP.

**Comment by lizziel** [comment]
My bad, I didn't realize this was metals simulation when I transferred here! @herizonarivelo, if you want the metals simulation in the long-term then you can officially add it to GCHP. Since this is a specialty simulation the support team is relying on the community to add it. The steps to do this are as follows:
1. Copy the template metals files located in the *.templates folders at [geoschem/run/GCClassic](https://github.com/geoschem/geos-chem/tree/main/run/GCClassic).
2. Put each one in the appropriate. *.templates files at [geoschem/run/GCHP](https://github.com/geoschem/geos-chem/tree/main/run/GCHP)
3. Adapt each file to be for GCHP by comparing a GC-Classic and GCHP template file for a different simulation, and also testing out in your local directory.
4. Add `ExtData.rc.templates/ExtData.rc.metals` to describe the input files
5. Update GCHP [createRunDir.sh](https://github.com/geoschem/geos-chem/blob/main/run/GCHP/createRunDir.sh) to add the metals simulation as an option. You can look at the [GC-Classic version of the file](https://github.com/geoschem/geos-chem/blob/main/run/GCClassic/createRunDir.sh) to see what special logic exists for metals.

If you do this and want to contribute it back to the standard model then you can create a PR on this repository.

**Comment by herizonarivelo** [comment]
Dear @yantosca and @lizziel , 

Thanks for your reply and your guidance on what to do to ensure that dry and wet deposits of SiF1 and SiF2 species are included in GCHP. 

I'll get back to you if I need anything but I believe that with the information you've given me, I hope it will work. 
Thanks a lot again!

**Comment by yantosca** [comment]
Thanks so much @herizonarivelo.  I'll close out this issue.

**Comment by herizonarivelo** [comment]
Dear @yantosca  and @lizziel ,

I followed @yantosca 's recommendations and still got error in return from my job.
@lizziel , I'm not doing a long-term metal simulation, However I need the dry and wet deposition for Silicone (Si) and also for other metal species such as Fe, Calcium... with my c48 fullchem simulation. 

So when I check my slurm out, it tell me that the error is the species Si_F1 and Si_F2 is not in the restart file.  So, do I need  a new restart file that contains the metals (Si, Ca,...) please? And if so, where can I find this restart file? For information I use the c48 restart file from this http://geoschemdata.wustl.edu/ExtData/GEOSCHEM_RESTARTS/GC_14.2.0/

Thank you in advance for your reply! And I'm sorry to reopen the discussion. 

I upload here my gc.log, the slurm and the HEMCO_Diag, ExtData, HEMCO_Diag that I modify.

[HEMCO_Config.rc.txt](https://github.com/user-attachments/files/17310786/HEMCO_Config.rc.txt)
[HEMCO_Diagn.rc.txt](https://github.com/user-attachments/files/17310789/HEMCO_Diagn.rc.txt)
[ExtData.rc.txt](https://github.com/user-attachments/files/17310790/ExtData.rc.txt)
[slurm-8026408.out.txt](https://github.com/user-attachments/files/17310791/slurm-8026408.out.txt)
[gchp.20060101_0000z.log.txt](https://github.com/user-attachments/files/17310792/gchp.20060101_0000z.log.txt)



Kind regards ;
Herizo

**Comment by herizonarivelo** [comment]
Thank you @yantosca  for your reply. There are still errors that occur and my simulation has crashed. To find out more about the error, I enabled DEBUG mode in the logging.yml for CAP.EXDATA and MAPL for root_level and level to get further details but still I don't understand why my simulation has stopped. It only shows a UCX error in my gchp.log but otherwise I can't see anything wrong. 

Is there something wrong that I have done in the ExtData? I added the HEMO_Config line from the metal simulation to GCHP and modified my HEMCO_Diag by comparing it with the HEMCO_Diag from the GCClassic (metal simulation). 
Thank you very much for your precious support. 
I put here the feedback of my job. 

[ExtData.rc.txt](https://github.com/user-attachments/files/17317433/ExtData.rc.txt)
[HEMCO_Diagn.rc.txt](https://github.com/user-attachments/files/17317434/HEMCO_Diagn.rc.txt)
[HEMCO_Config.rc.txt](https://github.com/user-attachments/files/17317435/HEMCO_Config.rc.txt)
[allPEs.log.txt](https://github.com/user-attachments/files/17317436/allPEs.log.txt)
[slurm-8027557.out.txt](https://github.com/user-attachments/files/17317437/slurm-8027557.out.txt)
[gchp.20060101_0000z.log.txt](https://github.com/user-attachments/files/17317438/gchp.20060101_0000z.log.txt)

**Comment by yantosca** [comment]
Thanks for the update @herizonarivelo.  I looked at your gchp*.log file and there are a bunch of errors like:
```console
[1728499161.738278] [n203:2682 :0]           sock.c:344  UCX  ERROR recv(fd=35) failed: Connection reset by peer
```
which I think is an MPI issue.  It could be caused by something in GCHP but I'm not sure what.  

I searched on the web and came across this post: https://github.com/openucx/ucx/discussions/8911.  They suggest adding `-x UCX_LOG_LEVEL_TRIGGER=error -x UCX_HANDLE_ERRORS=bt,freeze` to the `mpirun` command, which should generate additional error output.  Not sure if that will show any useful output that we can interpret but it's worth a shot.

**Comment by lizziel** [comment]
@herizonarivelo, there is traceback of the error in your slurm log file. The first line (meaning the source of the error) is this:
`pe=00000 FAIL at line=00639    ExtDataGridCompMod.F90                   <status=41>`

This means the problem is line 639 in file `ExtDataGridCompMod.F90`. You can find that file by doing `find . -name ExtDataGridCompMod.F90` from the top-level of the source code. Opening the file, it looks like the problem area is reading offset and scaling from `ExtData.rc`. Looking at your `ExtData.rc` I see you are missing offset and scaling for some of your entries, e.g.

```
NEI11_SURFACE_Al_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Al ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_As_F1 kg/m2/s N Y F%y4-01-01T00:00:00 As ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Ca_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Ca ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Cd_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Cd ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Fe_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Fe ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_K_F1  kg/m2/s N Y F%y4-01-01T00:00:00 K  ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Mg_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Mg ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Mn_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Mn ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Ni_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Ni ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Pb_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Pb ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Si_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Si ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Si_F2 kg/m2/s N Y F%y4-01-01T00:00:00 Si ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Ti_F1 kg/m2/s N Y F%y4-01-01T00:00:00 Ti ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
NEI11_SURFACE_Ti_F2 kg/m2/s N Y F%y4-01-01T00:00:00 Ti ./HcoDir/TRACE_METALS/v2021-06/USA/2011/%m2/NEI11_0.1x0.1_2011%m2%d2_all.nc
```

**Comment by herizonarivelo** [comment]
Dear all,

Many thanks for your comments. I'm sorry if I'm taking too long to write back.
My simulation is no longer stops when :
a). I disable the requires restart file i.e Require_Species_in_Restart=0 in setCommonRunSettings.sh
b). I have set the  SiF1 and SiF2 species as transported species in geochem_config.yml
c). I updated my HISTORY as I want with the dry and wet deposition of these species .

Now for my one day simulation test, I have the dry and wet depositions of  SiF1 and SiF2 species in my output netcdf. 

Thanks for any further advice.
Herizo Narivelo
